tools/cli/go-whisk/whisk/client.go

```diff
@@ -94,30 +94,39 @@ var DefaultObfuscateArr = []ObfuscateSet{
 
 func NewClient(httpClient *http.Client, config *Config) (*Client, error) {
 
+    if httpClient == nil {
+        httpClient = http.DefaultClient
+    }
+
     // Disable certificate checking in the dev environment if in insecure mode
+    var tlsConfig *tls.Config
     if config.Insecure {
         Debug(DbgInfo, "Disabling certificate checking.\n")
-        var tlsConfig *tls.Config
+        tlsConfig = &tls.Config{
+            InsecureSkipVerify: true,
+        }
         if config.Cert != "" && config.Key != "" {
-            if cert, err := tls.LoadX509KeyPair(config.Cert, config.Key); err == nil {
+            if cert, err := ReadX509KeyPair(config.Cert, config.Key); err == nil {
                 tlsConfig = &tls.Config{
                     Certificates: []tls.Certificate{cert},
                     InsecureSkipVerify: true,
                 }
             }
-        }else{
-            tlsConfig = &tls.Config{
-                InsecureSkipVerify: true,
-            }
         }
-
-        http.DefaultClient.Transport = &http.Transport{
-            TLSClientConfig: tlsConfig,
+    } else {
+        // Enable certificate checking in the environment for secure mode
+        Debug(DbgInfo, "Enabling certificate checking.\n")
+        if config.Cert != "" && config.Key != "" {
+            if cert, err := ReadX509KeyPair(config.Cert, config.Key); err == nil {
+                tlsConfig = &tls.Config{
+                    Certificates: []tls.Certificate{cert},
+                }
+            }
         }
     }
 
-    if httpClient == nil {
-        httpClient = http.DefaultClient
+    httpClient.Transport = &http.Transport{
```

Comment:
I think we have a different logic now
```
var tlsConfig *tls.Config
httpClient.Transport = &http.Transport{
    TLSClientConfig: tlsConfig,
}	
```
this will be the result when config.Insecure config.Cert and config.key are not defined.

Reply:
Yes, I just changed the PR by adding 
tlsConfig = &tls.Config {
       InsecureSkipVerify: config.Insecure,
}
tlsConfig will hve a default value based on the config.Insecure, no matter cert or key is available.